.github/workflows/reusable_build.yaml

```diff
@@ -31,52 +33,48 @@ jobs:
     steps:
-      - name: install_clang_and_tools
-        run: sudo apt update && sudo apt install -y clang clang-tools lld wget python3-pip python3-colcon-coveragepy-result python3-colcon-lcov-result lcov
-      # TODO: Remove this step when the incompatibility between libunwind14 and libundind dissapears https://github.com/open-rmf/rmf_traffic_editor/issues/439
-      - name: Horrible hack for libceres
-        run: |
-          apt-get remove -y --purge libc++-dev || true
-          apt-get remove -y --purge libc++abi-dev || true
-          apt-get remove -y --purge libunwind-14-dev || true
-          apt-get remove -y --purge libunwind-14-dev || true
-          apt-get remove -y --purge libunwind-dev || true
-          apt -y autoremove
-        if: ${{ matrix.ubuntu_distribution == 'jammy'}}
       - name: create_blacklist
         run: |
           mkdir -p ${{ github.workspace }}/
           touch ${{ github.workspace }}/blacklist.txt
           if [[ ${{ inputs.mixin }} == "asan" ]]; then
             echo "fun:*Eigen*" >  ${{ github.workspace }}/blacklist.txt
           fi
+      - name: Setup ROS
+        uses: ros-tooling/setup-ros@d8f32e519cd758bc3a3ffa0bf13ec77544aae30f
+      - name: install_tools
+        run: sudo apt update && sudo apt install -y ccache clang clang-tools lld wget python3-pip python3-colcon-coveragepy-result python3-colcon-lcov-result lcov ros-${{ matrix.ros_distribution }}-rmw-cyclonedds-cpp
       - name: set mixins
         id: set_mixins
         run: |
           case ${{ inputs.mixin }} in
             asan)
-                echo 'colcon_defaults={"build":{"mixin":["asan-gcc","lld"],"cmake-args":["-DCMAKE_BUILD_TYPE=Debug"],"executor":"sequential"}}' >> $GITHUB_OUTPUT
+                echo 'colcon_defaults={"build":{"mixin":["asan-gcc", "lld"],"cmake-args":["-DCMAKE_BUILD_TYPE=Debug"],"executor":"sequential"}}' >> $GITHUB_OUTPUT
                 ;;
             tsan)
-                echo 'colcon_defaults={"build":{"mixin":["tsan"],"cmake-args":["-DCMAKE_BUILD_TYPE=Debug"]}}' >> $GITHUB_OUTPUT
+                echo 'colcon_defaults={"build":{"mixin":["tsan", "lld"],"cmake-args":["-DCMAKE_BUILD_TYPE=Debug"]}}' >> $GITHUB_OUTPUT
                 ;;
             *)
-                echo 'colcon_defaults={"build": {"mixin": ["coverage-gcc", "lld" ]}}' >> $GITHUB_OUTPUT
+                echo 'colcon_defaults={"build": {"mixin": ["ccache", "coverage-gcc", "lld"]}}' >> $GITHUB_OUTPUT
                 ;;
           esac
+      - uses: actions/cache@v3
+        with:
+          path: ~/.cache/ccache
+          key: ccache
       - name: build_and_test
         uses: ros-tooling/action-ros-ci@v0.3
         env:
           LANG: en_US.UTF-8
           CC: clang -fsanitize-blacklist=${{ github.workspace }}/blacklist.txt
           CXX: clang++ -fsanitize-blacklist=${{ github.workspace }}/blacklist.txt
+          RMW_IMPLEMENTATION: rmw_cyclonedds_cpp
```

Comment:
@luca-della-vedova could you add a comment here about the asan issue with fastdds and hence the need to set the RMW to cyclonedds?

Reply:
https://github.com/open-rmf/rmf_ci_templates/pull/10/commits/22aa0940ae2bc002f2aa3580b06db2d53a7bf73c